src/org/ods/orchestration/util/MROPipelineUtil.groovy

```diff
@@ -324,6 +325,15 @@ class MROPipelineUtil extends PipelineUtil {
                         } else {
                             this.logger.debug("Repo '${repo.id}' is of type ODS Code Component. Nothing to do in phase '${name}' for target environment '${targetEnvToken}'.")
                         }
+                    } else if (repo.type?.toLowerCase() == PipelineConfig.REPO_TYPE_ODS_INFRA) {
+                        this.steps.stage('ODS Infrastructure Component') {
+                            // FIXME: consult SMEs which STAGE we should use: BUILD or DEPLOY
```

Comment:
@michaelsauter asking for your support here. Currently, the blocks for every repo type look differently. Unsure about how the block for infrastructure components (e.g. Cloud services) should look like. Want to advise?

Reply:
I am not really sure here. In the end, it is up to the "repo type" to decide what to do and when to do it. It depends on what fits best to the type.

It looks like your are executing the `Jenkinsfile` here, which presumably deploys infrastructure? Then moving it to the `DEPLOY` step would sound logical to me.

Note that for the `ods` type, we build+deploy in `BUILD` when targeting DEV, and deploy in `DEPLOY` when targeting QA/PROD. This is not really great, and I'd love to split this when targeting DEV in the future (so that deployment also happens in `DEPLOY` for DEV).

Reply:
In the end, we need to make sure that… the Jenkinsfile is executed for D/Q/P. Would you suggest to change the condition so that it is executed both in `build` and `deploy`? Or is `deploy` enough? How should this look like with the proper combination of `isAssembleMode` and `is PromotionMode`?

Reply:
Only execute once. If you want to run for D/Q/P, then it does not matter which "mode" it is - meaning the condition should just consider if the phase and the repo type.

Reply:
@michaelsauter so you say that the code is good to stay as it is and where it is to be triggered when deploying to D/Q/P?

Reply:
@metmajer Yes to me it looks like this will be executed for D/Q/P and can stay like it is. You might want to execute it in the `DEPLOY` phase instead, but this is a matter of preference. I would tend to move it to `DEPLOY` but only my preference - both ways will work.

Reply:
@michaelsauter ah, now I got you. Yes, `DEPLOY` is where it should be (as you can see from my comment further above, I had assumed it was already). Cheers!!

Reply:
Resolved via https://github.com/opendevstack/ods-jenkins-shared-library/pull/388/commits/9cf62cedc9a7c1f9087348eeef5f5bdff0bcc024